rasa/core/test.py

```diff
@@ -600,47 +570,43 @@ def _collect_action_executed_predictions(
+    previously_predicted_action_unlikely_intent = False
 
     if circuit_breaker_tripped:
         prediction = PolicyPrediction([], policy_name=None)
         predicted_action = "circuit breaker tripped"
     else:
-        action, prediction = processor.predict_next_action(partial_tracker)
-        predicted_action = action.name()
-
-        policy_entity_result = _get_e2e_entity_evaluation_result(
-            processor, partial_tracker, prediction
+        predicted_action, prediction, policy_entity_result = _run_action_prediction(
+            processor, partial_tracker, expected_action
         )
 
-        if (
-            prediction.policy_name
-            and predicted_action != expected_action
-            and _form_might_have_been_rejected(
-                processor.domain, partial_tracker, predicted_action
+    predicted_action_unlikely_intent = predicted_action == ACTION_UNLIKELY_INTENT_NAME
+    if predicted_action_unlikely_intent and predicted_action != expected_action:
+        partial_tracker.update(
+            WronglyPredictedAction(
+                predicted_action,
+                expected_action_text,
+                predicted_action,
+                prediction.policy_name,
+                prediction.max_confidence,
+                event.timestamp,
+                metadata=prediction.action_metadata,
             )
-        ):
-            # Wrong action was predicted,
-            # but it might be Ok if form action is rejected.
-            emulate_loop_rejection(partial_tracker)
-            # try again
-            action, prediction = processor.predict_next_action(partial_tracker)
-
-            # Even if the prediction is also wrong, we don't have to undo the emulation
-            # of the action rejection as we know that the user explicitly specified
-            # that something else than the form was supposed to run.
-            predicted_action = action.name()
-
-    predicted_warning = action_executed_eval_store.predicted_warning(predicted_action)
-    if not predicted_warning:
+        )
+        previously_predicted_action_unlikely_intent = True
+        predicted_action, prediction, policy_entity_result = _run_action_prediction(
+            processor, partial_tracker, expected_action
+        )
+        action_executed_eval_store.add_to_store(
+            action_predictions=[predicted_action], action_targets=[expected_action]
+        )
+    else:
         action_executed_eval_store.add_to_store(
             action_predictions=[predicted_action], action_targets=[expected_action]
         )
 
-    has_prediction_target_mismatch = (
-        action_executed_eval_store.has_prediction_target_mismatch()
-    )
-    if has_prediction_target_mismatch or (
-        predicted_warning and predicted_action != expected_action
+    if action_executed_eval_store.has_prediction_target_mismatch() or (
+        predicted_action_unlikely_intent and predicted_action != expected_action
```

Comment:
Isn't this part now the equivalent of `previously_predicted_action_unlikely_intent`?

Reply:
No, because we make another prediction => `predicted_action` gets overwritten